warehouse/query-core/src/main/java/datawave/query/function/DocumentProjection.java

```diff
@@ -38,40 +38,43 @@ public class DocumentProjection implements DocumentPermutation {
     private boolean trackSizes = true;
 
     public DocumentProjection() {
-        this(false, false);
+        this.includeGroupingContext = false;
+        this.reducedResponse = false;
+        this.projection = new Projection();
     }
 
     public DocumentProjection(boolean includeGroupingContext, boolean reducedResponse) {
-        this(includeGroupingContext, reducedResponse, true);
+        this.includeGroupingContext = includeGroupingContext;
+        this.reducedResponse = reducedResponse;
+        this.projection = new Projection();
     }
 
-    public DocumentProjection(boolean includeGroupingContext, boolean reducedResponse, boolean trackSizes) {
+    public DocumentProjection(boolean includeGroupingContext, boolean reducedResponse, boolean trackSizes, Set<String> projections,
+                    Projection.ProjectionType projectionType) {
         this.includeGroupingContext = includeGroupingContext;
         this.reducedResponse = reducedResponse;
-        this.projection = new Projection();
+        this.projection = new Projection(projections, projectionType);
         this.trackSizes = trackSizes;
     }
 
-    /**
-     * Set the delegate {@link Projection} with the fields to include
-     *
-     * @param includes
-     *            the set of fields to include
-     */
+    @Deprecated
     public void setIncludes(Set<String> includes) {
         this.projection.setIncludes(includes);
     }
 
-    /**
-     * Configure the delegate {@link Projection} with the fields to exclude
-     *
-     * @param excludes
-     *            the set of fields to exclude
-     */
+    public DocumentProjection(Set<String> projections, Projection.ProjectionType projectionType) {
```

Comment:
please group constructors together at the top of the class.

Reply:
you got it.